conifer/backends/vhdl/writer.py

```diff
@@ -49,6 +49,7 @@ def __init__(self, ensembleDict, config, metadata=None):
         # Convert the floating point values to integers
         tree.threshold_int = np.array([self._fp_converter.to_int(x) for x in tree.threshold])
         tree.value_int = np.array([self._fp_converter.to_int(x) for x in tree.value])
+        assert sum([(sum(weights)) for weights in tree.weight]) == self.max_depth + 1, f'Oblique splits are not supported by the VHDL backend, please use the hls backend'
```

Comment:
For robustness and maintainability, I think this logic could be moved up to the model base class wrapped as a function called `isOblique()` or similar.